Integrations/src/main/java/io/deephaven/integrations/learn/Gatherer.java

```diff
@@ -0,0 +1,223 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.db.v2.sources.ColumnSource;
+
+public class Gatherer {
+
+    /**
+     * Copy data from a table into a 2d tensor of Booleans.
+     *
+     * @param indexSet indices of the rows of the table to put into the tensor
+     * @param columnSources columns of data to put into the tensor
+     * @return contiguous RAM allocated for the tensor. When a numpy tensor is passed in for this argument, jpy will
+     *         handle passing the memory reference as a 1d java array here.
+     */
+
+    public static Boolean[] booleanTensorBuffer2D(final IndexSet indexSet, final ColumnSource<?>[] columnSources) {
```

Comment:
1) I think you want this to return `boolean[]`, since `Boolean[]` is going to be a mess in python and won't work with numpy -- for sure.  Should this throw an exception on null values?
2) I suggest naming these `tensorBuffer2D<type>` since they will then be ordered better in an IDE.